Fix Help Information in the Instrument Viewer

cailafinn opened the pull request:
**Description of work.**
Directed the instrument viewer to the correct documentation (for workbench, rather than plot) and have it open in the qthelp viewer rather than a web page.

Also updated the `.rst` documentation to reflect the actual functionality present in the GUI. 

**To test:**
1. Delete `doctrees`
2. Build the `docs-qthelp` target. 
3. Build and launch workbench
4. Open the instrument viewer on a workspace
5. Click the ``?`` button
6. The workbench help should appear with the new docs.
7. Check the docs for formatting, spelling, and grammar. 

Fixes #33437

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Use qthelp workbench docs in the instrument viewer

RE #33437

Commit:
Add documentation for pick tab tools

RE #33437

Commit:
Complete Draw Tab Documentation

RE #33437

Commit:
Add Instrument tab information

RE #33437

Commit:
Tidy Python Control section

RE #33437

Commit:
Improve grammar of existing documantation

RE #33437

Commit:
Add a release note

RE #33437

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -192,5 +332,5 @@ To start the app outside ``MantidWorkbench``, use the following code snippet as
 
-NOTE: If the instrument view app is launched outside ``MantidWorkbench`` on RHEL_7, the user need click on the viewing widget to get it rendered properly.
-For other OS, the instrument viewing widget should display the instrument properly upon launching.
+NOTE: If the Instrument Viewer is launched outside outside of Workbench on RHEL_7, the user will need to click on the viewing widget to allow it to render correctly.
```
```suggestion
NOTE: If the Instrument Viewer is launched outside of Workbench on RHEL_7, the user will need to click on the viewing widget to allow it to render correctly.
```

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
 --------------
 Many aspects of the instrument view can be controlled from Python.
-To use the Python interface for InstrumentViewer, use the following code to import the necessary libraries
+To use the Python interface for the InstrumentViewer, use the following code to import the necessary libraries:
```
```suggestion
To use the Python interface for the Instrument Viewer, use the following code to import the necessary libraries:
```

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -7,61 +7,58 @@ Instrument Viewer Widget
 --------------------
-The instrument view window can be obtained by right clicking on the workspace of interest and selecting "Show instrument".
+The instrument view window can be obtained by right clicking on the workspace of interest and selecting ``Show Instrument``.
 A 2D projection of the detector arrangement will be visible with each detector pixel color coded depending on the integrated number of counts in its corresponding spectrum, as in the image below.
```
A mixture of color and colour is used in this document. Whilst matplotlib uses the American spelling and so specific references to matplotlib features need to be spelled correctly in relation to that I think we need some consistency when using these words outside that context. In this case can we change `detector pixel color` to `detector pixel colour`?

sf1919 replied:
Are we not changing this one?

sf1919 replied:
Ah I see we're going with American spelling throughout so ignore my comment here.

ConorMFinn replied:
Yeah, it'd just end up horrible with the mix of mpl and our own code that uses the US spelling.

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -7,61 +7,58 @@ Instrument Viewer Widget
+There are other mouse and key board controls, which are listed at the bottom of the instrument view window.
 
-The control panel of the instrument view has four tabs: Render, Pick, Mask, and Instrument Tree.
+The control panel of the instrument view has four tabs: Render, Pick, Mask, and Instrument.
```
```suggestion
The control panel of the instrument view has four tabs: ``Render``, ``Pick``, ``Mask``, and ``Instrument``.
```

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -7,61 +7,58 @@ Instrument Viewer Widget
 The top-most combo-box control allows the user to select the way the instrument is projected onto the screen.
-The default setting is "Full 3D" which gives a 3D view of the instrument in an orthogonal projection.
+The default setting is ``Full 3D`` which gives a 3D view of the instrument in an orthogonal projection.
 The other options are "unwrapped views" projecting the instrument onto a curved surface and then unwrapping it onto the screen.
```
Should unwrapped views be formatted differently? e.g. ``unwrapped views``

ConorMFinn replied:
Nah, it's just a turn of phrase here, rather than referencing anything on the GUI. That sentence is poor though, so I'll rewrite it anyway

sf1919 replied:
I think for turns of phrase we generally use single quotes - certainly makes it less confusing that double quotes which potentially could have been used accidentally instead of the code formatting...

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -7,61 +7,58 @@ Instrument Viewer Widget
-Clicking and dragging the upper half of the bar changes the maximum while the lower half modifies the minimum.
-Mantid comes with a number of color map files and these can be loaded by selecting the "Display Settings"->"Color Map".
+The axis also defines the minimum and maximum values, which can be edited using the text boxes below and above the colour bar.
+Mantid comes with a number of color map files and these can be loaded by selecting the ``Display Settings`` -> ``Color Map``.
```
Not sure which spelling of colour/color to use here.

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+
+.. |edit| image:: ../images/PickTabEditButton.png
+
+.. image:: ../images/Workbench/InstrumentViewer/DrawShapesExample.jpg
```
This image could do with being centred. It could also do with some space between the text and the image.

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+
+.. |tubeBank| image:: ../images/PickTabPickTubeButton.png
+
+.. image:: ../images/Workbench/InstrumentViewer/PickTubeContextMenu.png
```
The image could do with being centred.

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+
+.. |singleCrystal| image:: ../images/PickTabAddPeakButton.png
+
+.. image:: ../images/Workbench/InstrumentViewer/SingleCrystalPeakExample.png
```
The image could do with being centred

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+
+.. |peakComparison| image:: ../images/PickTabCPCompare.png
+
+.. image:: ../images/Workbench/InstrumentViewer/PeakComparisonExample.png
```
The image could do with being centred

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+
+.. |peakAlign| image:: ../images/PickTabCPAlign.png
+
+.. image:: ../images/Workbench/InstrumentViewer/PeaksAlignmentExample.png
```
The image could do with being centred.

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+.. image:: ../images/Workbench/InstrumentViewer/DrawShapesExample.jpg
+
+The property browser displays the parameters of the selected shape. All sizes are given in the coordinates of the unwrapped surface the shapes are drawn on.
+For example, for a spherical surface it will be polar and azimuth angles in radians. The shapes are "attached" to the surface and move with it when the view is zoomed or moved.
```
```suggestion
For example, for a spherical surface it will be polar and azimuth angles in radians. The shapes are 'attached' to the surface and move with it when the view is zoomed or moved.
```

ConorMFinn replied:
Why does this need to be changed?

sf1919 replied:
I think single quotes for turn of phrase as discussed above.

ConorMFinn replied:
Is that a standard? I'd use double quotes for that kind of thing like I'd do with my fingers when speaking

sf1919 replied:
I guess its a personal preference thing so lets leave it as it is.

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+
+.. |zoom| image:: ../images/PickTabZoomButton.png
+
+.. image:: ../images/Workbench/InstrumentViewer/DrawPropertyBrowser.png
```
Can this be centred too?

sf1919 commented on docs/source/workbench/instrumentviewer.rst:
```diff
@@ -70,65 +67,208 @@ Pick Tab
+Clicking the ``Apply bin mask to view`` button will use the :ref:`MaskBins <algm-MaskBins>` algorithm mask the selected bins in the data workspace.
+
+
+.. image:: ../images/Workbench/InstrumentViewer/DrawBinMask.png
```
Could we have some space around the image? Feels crowded by the text.

Commit:
Improve grammar and naming consistency

RE #33437

Commit:
Centre images and give them some margins

Figures in rst files get some margin around them, whereas images don't.
For future, most images used in our docs should probably be figures
unless they're being unsed inline.

RE #33437

sf1919 reviewed (approved):
I am happy with the changes that have been made as a result of my earlier review.

This is a much needed revision of the Instrument Viewer documentation. The additional text and images make it far more useful and easier to understand. It's a good revision.
